examples/n8n-mcp/source/server.py

```
@@ -0,0 +1,189 @@
+
+        # Security Validation: Check against an Allow-list
+        executable = args[0]
+        if executable not in ALLOWED_COMMANDS:
+            return ExecuteResponse(
+                stdout="",
+                stderr=f"Forbidden command: '{executable}'. Only {list(ALLOWED_COMMANDS)} are allowed.",
+                exit_code=1
+            )
+
+        # Execute the command, always from the WORKING_DIR directory
+        process = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)